test/test.oauth2.ts

```diff
@@ -1421,6 +1426,23 @@ describe('oauth2', () => {
       assert.strictEqual(params.client_id, 'overridden');
     });
 
+    it('getToken should use basic header auth if provided in options', async () => {
```

Comment:
can you add tests for the none and post type and a default case?

Reply:
Yeah, no problem.